server/utils/appInsightsUtils.ts

```diff
@@ -1,13 +1,43 @@
 import type { TelemetryClient } from 'applicationinsights'
-import { DistributedTracingModes, defaultClient, setup } from 'applicationinsights'
+import { Contracts, DistributedTracingModes, defaultClient, setup } from 'applicationinsights'
+import type { EnvelopeTelemetry } from 'applicationinsights/out/Declarations/Contracts'
 
 import { buildNumber, packageData } from '../applicationVersion'
+import type { User } from '../data'
+
+type ContextObjectWithUser = {
+  res?: {
+    locals?: {
+      user: Partial<User>
+    }
+  }
+}
+
+export type ContextObject = Record<string, ContextObjectWithUser> | undefined
```

Comment:
A `ContextObject` is an object with key/value pairs with strings pointing to `ContextObjectWithUser`s? So a context object includes context objects as its keys' values?

Below your argument of `contextObjects` (plural) is a `ContextObject` - perhaps this type should be `ContextObjects`? Are there going to be multiple `ContextObjectWithUser`s in any `ContextObject`s?